cdap-ui/app/cdap/components/AbstractWidget/InputKeyValueDropdown/index.tsx

```diff
@@ -0,0 +1,146 @@
+  IMultiRowProps,
+  IMultiRowWidgetProps,
+} from 'components/AbstractWidget/AbstractMultiRowWidget';
+import { IStageSchema } from 'components/AbstractWidget';
+import KeyValueDropdownRow, {
+  IDropdownOption,
+  OrderingEnum,
+} from 'components/AbstractWidget/KeyValueDropdownWidget/KeyValueDropdownRow';
+
+import ThemeWrapper from 'components/ThemeWrapper';
+import { WIDGET_PROPTYPES } from 'components/AbstractWidget/constants';
+import { objectQuery } from 'services/helpers';
+
+interface IInputKeyValueDropdownWidgetProps extends IMultiRowWidgetProps {
+  'key-placeholder'?: string;
+  'kv-delimiter'?: string;
+  dropdownOptions: IDropdownOption[];
+  delimiter?: string;
+  showDelimiter?: boolean;
+  extraConfig?: any;
+}
+
+interface IInputKeyValueDropdownProps extends IMultiRowProps<IInputKeyValueDropdownWidgetProps> {}
+
+interface IField {
+  name: string;
+  type: string[];
+}
+
+class InputKeyValueDropdownWidgetView extends AbstractMultiRowWidget<IInputKeyValueDropdownProps> {
+  public getFields(schemas: IStageSchema[], allowedTypes: string[]) {
+    let fields = [];
+    if (!schemas || schemas.length === 0) {
+      return fields;
+    }
+    const stage = schemas[0];
+
+    try {
+      const unparsedFields = JSON.parse(stage.schema).fields;
+
+      if (unparsedFields.length > 0) {
+        fields = unparsedFields
+          .filter((field: IField) => this.containsType(field.type, allowedTypes))
+          .map((field: IField) => field.name);
+      }
+    } catch {
+      // tslint:disable-next-line: no-console
+      console.log('Error: Invalid JSON schema');
+    }
+    return fields;
+  }
+
+  // Function that checks if types contains a type that is in allowedTypes
+  // This is meant to handle nullable fields since a nullable string type is
+  // presented as ['string','null'].
+  public containsType(types: string[], allowedTypes: string[]) {
+    if (allowedTypes.length === 0) {
+      return true;
+    }
+
+    return allowedTypes.includes(this.extractType(types));
+  }
+
+  public extractType(types) {
+    let value = types;
+    if (types instanceof Array) {
+      if (types.length === 1) {
+        value = types[0];
+      } else if (types.length === 2 && types.includes('null')) {
+        value = types.indexOf('null') === 0 ? types[1] : types[0];
+      }
+    }
+
+    if (typeof value === 'object') {
+      value = value.logicalType || value;
+    }
+    return value;
+  }
+
+  public renderRow = (id, index) => {
+    const keyPlaceholder = objectQuery(this.props, 'widgetProps', 'key-placeholder');
+    const kvDelimiter = objectQuery(this.props, 'widgetProps', 'kv-delimiter');
+    const dropdownOptions = objectQuery(this.props.extraConfig, 'inputSchema');
+    const fieldValues = this.getFields(dropdownOptions, []);
+
+    return (
+      <KeyValueDropdownRow
+        key={id}
+        value={this.values[id].value}
+        id={id}
+        index={index}
+        onChange={this.editRow}
```

Comment:
Since we flip the ordering of the widget, the component ultimately returns `'key:value'` as string. To give an example,

- User configures the widget. Selects 'column1' from the input schema
- In the pivot textbox enters `v1, v2, v3`
- On change the component sets the value as `v1,v2,v3:column1`.

If we are flipping the ordering of the component we should handle the onchange as well if we expect the value to be in the same ordering, in this case `column1:v1, v2, v3`